From c45a49bacfa335a7e0dbd0ba144a25a1bfe9cb9b Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Thu, 22 Mar 2018 12:00:46 +0200
Subject: [PATCH 120/706] enetc: Update PSIVLANR to new addr starting from
 ENETC BG v88

Conforms ENETC BG v88 (BOM 62.5), cca. m0169+ simu model.

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit b1f4ea5b9ae31f0a758d8453a2bd12ed311c8fb2)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 6 +++---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 5 +++--
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 21b8435a3d6b..b39ab6b3b2fd 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -186,9 +186,9 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_VLAN_TYPE_S	BIT(1)
 #define ENETC_PVCLCTR_OVTPIDL(bmp)	((bmp) & 0xff) /* VLAN_TYPE */
 
-#define ENETC_PSIIVLANR(n)	(0x00210 + (n) * 4) /* n = SI index */
-#define ENETC_PSIIVLAN_EN	BIT(31)
-#define ENETC_PSIIVLAN_SET_QOS(val)	((u32)(val) << 12)
+#define ENETC_PSIVLANR(n)	(0x00240 + (n) * 4) /* n = SI index */
+#define ENETC_PSIVLAN_EN	BIT(31)
+#define ENETC_PSIVLAN_SET_QOS(val)	((u32)(val) << 12)
 #define ENETC_PCAPR0	0x00900
 #define ENETC_PCAPR0_RXBDR(val)	((val) >> 24)
 #define ENETC_PCAPR0_TXBDR(val)	(((val) >> 16) & 0xff)
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index acc17995a28e..b80b2672e12c 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -73,8 +73,9 @@ static void enetc_set_isol_vlan(struct enetc_hw *hw, int si, u16 vlan, u8 qos)
 	u32 val = 0;
 
 	if (vlan)
-		val = ENETC_PSIIVLAN_EN | ENETC_PSIIVLAN_SET_QOS(qos) | vlan;
-	enetc_port_wr(hw, ENETC_PSIIVLANR(si), val);
+		val = ENETC_PSIVLAN_EN | ENETC_PSIVLAN_SET_QOS(qos) | vlan;
+
+	enetc_port_wr(hw, ENETC_PSIVLANR(si), val);
 }
 
 static int enetc_mac_addr_hash_idx(const u8 *addr)
-- 
2.17.1

